google/cloud/aiplatform/metadata/metadata_store.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Optional
+
+import logging
+from google.auth import credentials as auth_credentials
+from google.api_core import exceptions
+
+from google.cloud.aiplatform import base, initializer
+from google.cloud.aiplatform_v1beta1.types import metadata_store as gca_metadata_store
+from google.cloud.aiplatform_v1beta1.services.metadata_service import (
+    client as metadata_service_client,
+)
+
+
+class MetadataStore(base.AiPlatformResourceNounWithFutureManager):
+    """Managed MetadataStore resource for AI Platform"""
+
+    client_class = metadata_service_client.MetadataServiceClient
+    _is_client_prediction_client = False
+    _resource_noun = "metadataStores"
+    _getter_method = "get_metadata_store"
+    _delete_method = "delete_metadata_store"
+
+    def __init__(
+        self,
+        metadata_store_name: Optional[str] = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing MetadataStore given a MetadataStore name or ID.
+
+        Args:
+            metadata_store_name (str):
+                Optional. A fully-qualified MetadataStore resource name or metadataStore ID.
+                Example: "projects/123/locations/us-central1/metadataStores/my-store" or
+                "my-store" when project and location are initialized or passed.
+                If not set, metadata_store_name will be set to "default".
+            project (str):
+                Optional project to retrieve resource from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional location to retrieve resource from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+
+        """
+
+        super().__init__(
+            project=project, location=location, credentials=credentials,
+        )
+        self._gca_resource = self._get_gca_resource(resource_name=metadata_store_name)
+
+    @classmethod
+    def create(
```

Comment:
Not required but you have the option to make this optionally asynchronous using the following pattern. Right now this would only be useful if you didn't want to block on the `init` call. But calling asynchronously has the downside of this potentially failing and the user not realizing until they make a Experiment tracking call ie: log_param/log_metric:

```python
@classmethod
def create(..., sync=True):
   # synchronously validate arguments and capture an require global configs, create client

@classmethod
@base.optional_sync()
def _create(..., sync=True):
  # make api call and return MetadataStore object
```

Here an example of how `Endpoint.create` handles this:
`create`: https://github.com/googleapis/python-aiplatform/blob/dev-test/google/cloud/aiplatform/models.py#L110
`_create`: https://github.com/googleapis/python-aiplatform/blob/dev-test/google/cloud/aiplatform/models.py#L197